src/dotnet/commands/dotnet-build/ProjectGraphCollector.cs

```diff
@@ -43,7 +43,7 @@ private ProjectGraphNode TraverseProject(ProjectDescription project, IDictionary
                 foreach (var dependency in project.Dependencies)
                 {
                     LibraryDescription libraryDescription;
-                    if (lookup.TryGetValue(dependency.Name, out libraryDescription))
+                    if ((lookup.TryGetValue(dependency.Name, out libraryDescription)) && (!libraryDescription.Identity.Name.Equals(project.Identity.Name)))
```

Comment:
Nice!

I'm not too familiar with this part of the code so tell me if I'm way off here. From looking at the `if` statement it seems like this would also fail if you had a package reference (not `target: project`) with the same name as the current project. 

Is that the case? If so should we scope it only to fail for `target: project` references?